packages/modals/src/elements/Close.spec.tsx

```diff
@@ -25,4 +26,34 @@ describe('Close', () => {
 
     expect(getByTestId('close')).toBe(ref.current);
   });
+
+  describe('functionality', () => {
+    it('sets isCloseButtonPresent when mounting and unmounting', () => {
```

Comment:
Since the source code changes involve the end user UI, I think the test should resemble how users interact with the component.

I was going over the [testing-lib docs](https://testing-library.com/docs/guiding-principles) and it looks like `setCloseButtonPresent` is similar to an instance method. The guide suggests asserting against the DOM, rather than against the instance method or callbacks. The tests are less brittle when they target the end user experience rather than implementation details when testing UI. For example, if an implementation detail changed (e.g. rename `setCloseButtonPresent`), then the current test breaks.

I think the following test would better align with testing-lib's docs and less brittle:

```tsx
describe('DrawerModal', () => { // DrawerModa.spec.tsx
  it('has horizontal padding when a DrawerModal.Close is present', () => {
    render(
      <DrawerModal isOpen>
        <DrawerModal.Header>Header</DrawerModal.Header>
        <DrawerModal.Close />
      </DrawerModal>
    );

    expect(screen.getByText('Header')).toHaveStyleRule('padding-right', '56px');
  });

  it('does not have horizontal padding when a DrawerModal.Close is absent', () => {
    render(
      <DrawerModal isOpen>
        <DrawerModal.Header>Header</DrawerModal.Header>
      </DrawerModal>
    );

    expect(screen.getByText('Header')).not.toHaveStyleRule('padding-right');
  });
});
```

Note: The RTL behavior is asserted in spec files under `/styled` — so no change there.

Reply:
Good point @hzhu, I will update these tests.